superset-frontend/src/explore/components/ExploreViewContainer.jsx

```diff
@@ -404,7 +414,8 @@ function ExploreViewContainer(props) {
         />
       )}
       <Resizable
-        defaultSize={{ width: 300 }}
+        onResize={onResize}
+        defaultSize={{ width: exploreSidebarWidth }}
```

Comment:
I think the dataset pane and the controls pane should have separate widths. You might want to create a pair of small getter/setter functions to handle saving and reading both widths in one key safely.

```ts
enum StorageKey  {
   sidebarWidth = "explore_sidebar_widths"
}

function getSidebarWidths() {
  try {
     return localStorage.getItem(StorageKey.siderbarWidth).split(":");
  } catch {}
  return [300, 320]
}

function setSidebarWidths(widths: number[]) {
  try {
    localStorage.setItem(StorageKey.siderbarWidth, widths.join(":"));
  } catch {}
}
```

Reply:
I noticed SQL Lab is using localstorage without a try/catch safety net. This PR makes me wonder if we should make an abstracted helper method, kind of like the "logging" method from `@superset-ui/core`. These methods, maybe `setStoredValue` and `getStoredValue, could try local storage, then try setting a cookie, and take an optional callback for the `catch` block. I don't think we need to get that crazy with this PR, but wonder if folks think it's an idea worth ticketing separately.

Reply:
I always do that in my other projects. Here's an example file with typing for storage keys:

```ts
import { User } from 'graphql/job.graphql';

/**
 * Local storage manager
 */
export function safeJsonParse(jsonString: string | null) {
  if (jsonString === null) return null;
  try {
    return JSON.parse(jsonString);
  } catch {
    return null;
  }
}

export type StorageKey = 'persistedStates' | 'hasWelcomed';

export type StorageValues = {
  persistedStates: Record<string, any> | null;
  hasWelcomed: boolean;
};

export const defaultStorageValues = {
  userIdMap: null,
  persistedStates: null,
  hasWelcomed: false,
};

export function entries() {
  Object.values(Storage).forEach(key => {
    return [key, getItem(key)];
  });
}

export function getItem<K extends StorageKey>(key: K): StorageValues[K] | null {
  return safeJsonParse(localStorage.getItem(key)) ?? defaultStorageValues[key];
}

export function setItem(key: StorageKey, value: string | number | object) {
  return localStorage.setItem(key, JSON.stringify(value));
}
```